### ISSUE TITLE
Article contains wrongful information

### ISSUE THREAD

**Issue by Saborion** [dotnet/docs#22704]
The last sentence in the following section is a bit misinformative:

"_As before, you must initialize the list object to a non-null value or the compiler issues a warning. There are no null checks in the second overload of AddQuestion because they aren't needed: You've declared that variable to be non-nullable. **Its value can't be null.**_"

I would argue that the value most definitely can be null; the compiler issuing a warning is not the same as "can't be null". I suspect the article is full or similar statements / code constructs.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b991cba9-efd4-68ae-283a-b575054b4907
* Version Independent ID: 793f790b-e2aa-e292-b12b-f4ba9763f172
* Content: [Design with nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/nullable-reference-types?source=docs)
* Content Source: [docs/csharp/tutorials/nullable-reference-types.md](https://github.com/dotnet/docs/blob/master/docs/csharp/tutorials/nullable-reference-types.md)
* Product: **dotnet-csharp**
* Technology: **csharp-null-safety**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by Youssef1313** [comment]
I agree given that it's a public API. Consumers may not even have opt-ed in NRT feature.

**Comment by BillWagner** [comment]
Thanks for raising this @Saborion 

Let me think about how to frame this better. We do overstate it, but I want to make sure we don't over correct.

**Comment by BillWagner** [comment]
This sentence (and possibly similar statements) should be softened:  Public APIs should verify arguments aren't null even when the type is a non-nullable reference type. Client code could have mistakes, or try to maliciously call the API incorrectly.
